### PR TITLE
Resolving divide by 0 error 

### PR DESCRIPTION
**Pull request by snigdhabanda**
Adding in error handling to `getPermitsRemaining()` in the case that a user sets the allowed `qps` to 0.

### CODE REVIEW

**Review by ean5533 — APPROVED** [review]
What, you don’t like having `NaN` permits remaining?

**Comment by snigdhabanda** [comment]
> What, you don’t like having `NaN` permits remaining?

bwamp bwamp bwamp

**Review by ean5533 — APPROVED** [review]

**Review by adrw — APPROVED** [review]

File: misk-core/src/main/kotlin/misk/sampling/RateLimiter.kt [hunk 1]
```diff
@@ -118,6 +118,7 @@ class RateLimiter private constructor(
     unit: TimeUnit,
     timeout: Long
   ): Long {
+    if (permitsPerSecond == 0L) return 0L
```
Comment (tyiu) [inline misk-core/src/main/kotlin/misk/sampling/RateLimiter.kt:121]: I think we could also short-circuit on negatives to make it a bit easier to understand on those edge cases. i.e.

```kotlin
if (permitsPerSecond <= 0L) return 0L
```

Tests for that scenario would be good as well.
